File: Android/shared/src/main/kotlin/shared/dto/objects/Customer.kt
@file:UseSerializers(UUIDSerializer::class)

package shared.dto.objects

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.enums.ContactType
import shared.dto.primitives.Token
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable
import shared.serializers.UUIDSerializer

object Customer {
    data object Parameters {
        @Serializable
        data class Registration(
            val contact: Internal.Contact? = null
        ) : Parametable()
    }

    data object Responses {
        @Serializable
        data class Registration(
            val accessToken: Token,
        ) : Responsable
    }
    data object Internal {
        @Serializable
        data class Contact(
            var value: String,
            var type: ContactType
        ) : Parametable()
    }
}

File: iOS-legacy/DTOs/Objects/Customer/Customer.swift
import Foundation

public enum Customer {
    public enum Parameters {}
    public enum Responses {}
    
    public enum Internal {}
}

public extension Customer.Parameters {
    
    struct Registration: Codable, Sendable {
        public var contact: Customer.Internal.Contact?
        
        public init(contact: Customer.Internal.Contact? = nil) {
            self.contact = contact
        }
    }
}

public extension Customer.Responses {
    
    struct Registration: Codable, Sendable {
        public var accessToken: Token
        
        public init(accessToken: Token) {
            self.accessToken = accessToken
        }
    }
}

public extension Customer.Internal {
    struct Contact: Codable, Sendable {
        public var value: String
        public var type: ContactType
        
        public init(value: String, type: ContactType) {
            self.value = value
            self.type = type
        }
    }
}
